### DISCUSSION TITLE
@Bindable does not resolve to correct @Perception.Bindable?

### DISCUSSION THREAD

**Discussion by VladOrackle** [pointfreeco/swift-perception#17] [General]
### Description

This problem is kind of strange, because I can safely write `@Bindable` even on iOS 15 but then the compiler argues that it can't find the model object. If I write the full `@Perception.Bindable` or use a typealias it works as expected. I suspect that it is an Xcode bug, but I might be wrong.

<img width="698" alt="Screenshot 2024-01-12 at 23 29 52" src="https://github.com/pointfreeco/swift-perception/assets/61315957/2264e338-6608-4fbf-8ab5-a4d95d871bb2">
<img width="689" alt="Screenshot 2024-01-12 at 23 30 05" src="https://github.com/pointfreeco/swift-perception/assets/61315957/78f70732-c923-4a95-a99c-d39de179fa98">
<img width="711" alt="Screenshot 2024-01-12 at 23 30 42" src="https://github.com/pointfreeco/swift-perception/assets/61315957/2fb01c97-d153-436b-92db-b5b82ece95bd">

### Checklist

- [X] I have determined whether this bug is also reproducible in a vanilla SwiftUI project.
- [X] If possible, I've reproduced the issue using the `main` branch of this package.
- [X] This issue hasn't been addressed in an [existing GitHub issue](https://github.com/pointfreeco/swift-perception/issues) or [discussion](https://github.com/pointfreeco/swift-perception/discussions).

### Expected behavior

@Bindable works without warnings

### Actual behavior

@Bindable works without warnings only if we use the full name provided by the library

### Steps to reproduce

Use @Perceptible model and try to use @Bindable inside views body

### Perception version information

1.0.0

### Destination operating system

iOS 15

### Xcode version information

15.2

### Swift Compiler version information

```shell
swift-driver version: 1.87.3 Apple Swift version 5.9.2 (swiftlang-5.9.2.2.56 clang-1500.1.0.2.5)
```

**Comment by mbrandonw** [comment]
Hi @VladOrackle, this does seem to be some kind of Swift bug, and I'm not sure there is anything we can do to work around it.

FWIW, if you had use `@Bindable` on the property of the view rather than in the body, you would have at least gotten a clearer error message of how to fix it:

```swift
struct ContentView: View {
  // 🛑 'Bindable' is ambiguous for type lookup in this context
  @Bindable var model: CounterModel
 
  …
}
```

Since this isn't an issue with the library I am going to convert it to a discussion.
